include/deal.II/hp/dof_handler.h

```diff
@@ -691,10 +707,7 @@ namespace hp
   private:
 
     /**
-     * Copy constructor. I can see no reason why someone might want to use it,
-     * so I don't provide it. Since this class has pointer members, making it
-     * private prevents the compiler to provide it's own, incorrect one if
-     * anyone chose to copy such an object.
```

Comment:
This comment is old and a bit misleading.

There is actually a reason why we do *not* provide a copy constructor for large objects such as <code>dof_handler</code>). With this we force that no accidental copy is generated (which can easily happen e.g. by accidentally passing by value instead of by reference).

From a design perspective a <code>dof_handler</code> should only exist once for a given triangulation and finite element pair.

Further, the copy constructor is still a private member in your proposed pull request...

Why do you need it?